### DISCUSSION TITLE
Any plan to support Pest?

### DISCUSSION THREAD

**Discussion by Niush** [recca0120/vscode-phpunit#152] [Ideas]
Currently, this extension works with PHPUnit only. Is there any plan to support [Pest](https://pestphp.com/) as well?

Parsing the test files will be completely different, so that might be a bottleneck issue.

**Comment by recca0120** [comment]
Yes, this extension will support Pest.

**Reply by JaberWiki** [reply]
@recca0120  It's not working on PestPHP !

**Comment by recca0120** [comment]
https://github.com/user-attachments/assets/466c0fb0-cfb1-4c90-9319-bfc1a2b36c71

working on this

**Reply by JaberWiki** [reply]
> 2024-12-12.9.16.03.mov 
> working on this

Great job!

**Comment by Smef** [comment]
One thing I've noticed with this: If you run PHPUnit tests using Pest, the extension will succeed, but not always record that the test has been run in the test explorer window. It will continue to show that the tests are "not yet run" even if they have completed successfully.

What is extra weird, is that it seems to be dependent on the class name. If the name is two words, it works fine. If the name is three words, the test is never shown as run.

"DeleteTest" works fine, but "DeleteAccountTest" will always have the "not yet run" status. Just renaming the class seems to affect the results.

**Comment by Smef** [comment]
Here are the two outputs of the same test. The only difference is changing the class "DeleteAccountTest" to "DeleteAccount". The file itself doesn't actually need to be renamed. I think the issue may be with the `pest_qn` attribute which has the space in it.

```
##teamcity[testSuiteStarted name='Tests\Feature\DeleteAccountTest' locationHint='pest_qn://Delete Account (Tests\Feature\DeleteAccount)' flowId='91975']
##teamcity[testCount count='1' flowId='91975']
##teamcity[testStarted name='User accounts can be deleted' locationHint='pest_qn://Delete Account (Tests\Feature\DeleteAccount)::User accounts can be deleted' flowId='91975']
##teamcity[testFinished name='User accounts can be deleted' duration='37' flowId='91975']
##teamcity[testSuiteFinished name='Tests\Feature\DeleteAccountTest' flowId='91975']

  Tests:    1 passed (1 assertions)
  Duration: 0.21s
```


```
##teamcity[testSuiteStarted name='Tests\Feature\DeleteTest' locationHint='pest_qn://Delete (Tests\Feature\Delete)' flowId='93188']
##teamcity[testCount count='1' flowId='93188']
##teamcity[testStarted name='User accounts can be deleted' locationHint='pest_qn://Delete (Tests\Feature\Delete)::User accounts can be deleted' flowId='93188']
##teamcity[testFinished name='User accounts can be deleted' duration='33' flowId='93188']
##teamcity[testSuiteFinished name='Tests\Feature\DeleteTest' flowId='93188']

  Tests:    1 passed (1 assertions)
  Duration: 0.21s
```

**Comment by recca0120** [comment]
Pest has been fully supported for a while now (v1 through v4).

@Smef Regarding the class name mismatch issue you reported — this was a Pest v1-specific edge case where Pest stripped the `Test` suffix from the class name in its teamcity output (`DeleteAccount` instead of `DeleteAccountTest`), causing the test explorer to fail matching results back to test items.

This should no longer be an issue with Pest v2+ which uses a completely different output format (`file://` paths with `__pest_evaluable_` method encoding). If you are still experiencing this on a recent version of Pest, please open a new issue with reproduction steps.

Closing as resolved.
